everything_else/reproducing.m

```diff
@@ -269,9 +269,48 @@
 RHS=min(reshape(b1.*b2 + ba1 + 2*ba2 + 2*(1-b1).*(1-ba1) + 2*(1-b2).*(1-ba1) + 2*(1-b3).*(1-ba2) + 2*(1-ba2).*(1-ba1) - 2*(1-b3) - 1 + b1.*b3 - b2,4,[]));
 isequal(LHS,RHS);
 
-%% Pg. 30, PTR-CZW
-%% Pg. 31, Bit-flipping
-%% Pg. 32, SFR-ABCG-1
+%% PG 30
+z=[1 0; 0 -1];
+z1 = kron(z,eye(16));
+z2 = kron(kron(eye(2),z),eye(8));
+z3 = kron(kron(eye(4),z),eye(4));
+z4 = kron(kron(eye(8),z),eye(2));
+%% Alternatively, zlist = zgenerate(4);
+LHS=z1*z2*z3*z4;
+b=bgenerate(4);
+RHS=16*b(:,1).*b(:,2).*b(:,3).*b(:,4)-8*(b(:,1).*b(:,2).*b(:,3)+b(:,1).*b(:,2).*b(:,4)+b(:,1).*b(:,3).*b(:,4)+b(:,2).*b(:,3).*b(:,4));
```

Comment:
In previous cases within this same file, we define `b1=b(:,1)` then just use b1. This is a bit more tedious from a coding perspective, but it makes the RHS a lot easier to read.

Reply:
Fixed